modules/images/webp-uploads/generation.php

```diff
@@ -0,0 +1,250 @@
+<?php
+/**
+ * On demand generation functions.
+ *
+ * @package performance-lab
+ * @since n.e.x.t
+ */
+
+/**
+ * Generates images on the fly for the registered image sizes and available transformations.
+ *
+ * @since n.e.x.t
+ */
+function webp_uploads_generate_missing_images() {
+	if ( ! is_404() ) {
+		return;
+	}
+
+	$uploads_dir = wp_upload_dir();
+	$baseurl     = wp_parse_url( $uploads_dir['baseurl'], PHP_URL_PATH );
+
+	// Bail if the current request is not to an uploaded image.
+	if ( substr( $_SERVER['REQUEST_URI'], 0, strlen( $baseurl ) ) !== $baseurl ) {
+		return;
+	}
+
+	$matches = array();
+	$relpath = str_replace( $baseurl, '', $_SERVER['REQUEST_URI'] );
+
+	// Bail if the current request is not to a subsize.
+	if ( ! preg_match( '#/(.+)-(\d+)x(\d+)\.([a-zA-Z]+)$#', $relpath, $matches ) ) {
+		return;
+	}
+
+	$img_width  = intval( $matches[2] );
+	$img_height = intval( $matches[3] );
+
+	$size       = '';
+	$size_names = get_intermediate_image_sizes();
+	$sizes      = wp_get_additional_image_sizes();
```

Comment:
Why not use [`wp_get_registered_image_subsizes()`](https://developer.wordpress.org/reference/functions/wp_get_registered_image_subsizes/) instead? You could reduce a lot of the code here, also the lookup of width and height you're doing below.